x-pack/plugin/stack/src/javaRestTest/java/org/elasticsearch/xpack/stack/EcsDynamicTemplatesIT.java

```diff
@@ -0,0 +1,400 @@
+    }
+
+    private void verifyEcsMappings(String indexName) throws IOException {
+        final Map<String, Object> rawMappings = getMappings(indexName);
+        final Map<String, String> flatFieldMappings = new HashMap<>();
+        final Map<String, String> flatMultiFieldsMappings = new HashMap<>();
+        processRawMappingsSubtree(rawMappings, flatFieldMappings, flatMultiFieldsMappings, "");
+
+        Map<String, Map<String, Object>> shallowFieldMapCopy = new HashMap<>(ecsFlatFieldDefinitions);
+        logger.info("Testing mapping of {} ECS fields", shallowFieldMapCopy.size());
+        List<String> nonEcsFields = new ArrayList<>();
+        Map<String, String> fieldToWrongMappingType = new HashMap<>();
+        flatFieldMappings.forEach((fieldName, actualMappingType) -> {
+            Map<String, Object> expectedMappings = shallowFieldMapCopy.remove(fieldName);
+            if (expectedMappings == null) {
+                nonEcsFields.add(fieldName);
+            } else {
+                String expectedType = (String) expectedMappings.get("type");
+                if (actualMappingType.equals(expectedType) == false) {
+                    fieldToWrongMappingType.put(fieldName, actualMappingType);
+                }
+            }
+        });
+
+        Map<String, String> shallowMultiFieldMapCopy = new HashMap<>(ecsFlatMultiFieldDefinitions);
+        logger.info("Testing mapping of {} ECS multi-fields", shallowMultiFieldMapCopy.size());
+        flatMultiFieldsMappings.forEach((fieldName, actualMappingType) -> {
+            String expectedType = shallowMultiFieldMapCopy.remove(fieldName);
+            if (expectedType != null) {
+                // not finding an entry in the expected multi-field mappings map is acceptable: our dynamic templates are required to
+                // ensure multi-field mapping for all fields with such ECS definitions. However, the patterns in these templates may lead
+                // to multi-field mapping for ECS fields for which such are not defined
+                if (actualMappingType.equals(expectedType) == false) {
+                    fieldToWrongMappingType.put(fieldName, actualMappingType);
+                }
+            }
+        });
+
+        shallowFieldMapCopy.keySet()
+            .forEach(field -> logger.error("ECS field '{}' is not covered by the current ECS dynamic templates", field));
+        shallowMultiFieldMapCopy.keySet().forEach(field -> {
+            int lastDotIndex = field.lastIndexOf('.');
+            String parentField = field.substring(0, lastDotIndex);
+            String subfield = field.substring(lastDotIndex + 1);
+            logger.error("ECS field '{}' is expected to have a multi-field mapping with subfield '{}'", parentField, subfield);
+        });
+        fieldToWrongMappingType.forEach((fieldName, actualMappingType) -> {
+            String ecsExpectedType = (String) ecsFlatFieldDefinitions.get(fieldName).get("type");
+            logger.error(
```

Comment:
Either the assertions or the error logs should contain some hints on how to fix the situation. Something like change ecs-dynamic-mappings.json and add a dynamic template to map field {} to type {}.